mqtt/mqtt-bridge/src/persist/mod.rs

```diff
@@ -19,3 +22,13 @@ pub enum PersistError {
     #[error("Attempted to remove entry which does not exist")]
     RemovalForMissing,
 }
+
+// This might replace the PersistError or merge with it.
+#[derive(Debug, thiserror::Error)]
+pub enum StorageError {
+    #[error("RingBuffer error occurred {0}")]
```

Comment:
nit: what I did in my previous PR: I added `Caused by: {0}` into the error description in the `mqtt-bridge` crate. It'd be nice if you also did that for the code/error you added. it is fine if do it in the next PR.

```suggestion
    #[error("RingBuffer error occurred. Caused by: {0}")]
```